src/threading.c

```diff
@@ -333,6 +333,7 @@ static jl_value_t *ti_run_fun(jl_callptr_t fptr, jl_method_instance_t *mfunc,
 // lock for code generation
 jl_mutex_t codegen_lock;
 jl_mutex_t typecache_lock;
+jl_mutex_t typemap_lock;
```

Comment:
This should be a static lock in the right file and not in header unless/until this is needed in multiple files